### ISSUE TITLE
Avoid needlessly bumping version requirements of dependencies

### ISSUE THREAD

**Issue by mattias-p** [zonemaster/zonemaster#1022]
Today the release process instruction prescribes that CLI and Backen always bump the version requirement for Engine to the latest release. In most release cycles we make updates to Engine as well as CLI and Backend.

Sometimes we need to bump the version requirement for Engine in CLI and/or Backend. In those cases we still should bump the version requirement. I.e.:
* When we make breaking changes in Engine such that aren't compatible with the old version of CLI and/or Backend.
* When we update CLI and/or Backend to make use of new features in Engine that aren't available at the current version requirement.

When neither of the above cases are true we have reason to believe that the new CLI and/or Backend versions are compatible with previous versions of Engine down to and including the old Engine version requirement. In these cases we have the opportunity to not bump the Engine version requirement to the new version. I believe we should take this opportunity. I can see two scenarios where this choice would make a difference.
1. If we do bump the version requirement without technical reasons and a bug slips into the new Engine version users simply won't be able to use the new version of CLI and/or Backend until we make a new release.
2. If we don't bump the version requirement and an incompatibility slips into the new CLI version users have the option to upgrade Engine to be able to use the new CLI version.

AFAICT this is a clear win for not bumping the Engine version requirement.

**Comment by matsduf** [comment]
I disagree to the suggested change.

* We test CLI (Backend) with the latest version of Engine and Zonemaster-LDNS. By also requiring the latest version we make sure that the users use the combination that we have tested, which increases the chance that things do go wrong.
* If there is a bug in Engine or Zonemaster-LDNS we will have to fix it anyway because that will create problem for new users.
* Power users can still use CLI or Backend with an older version of Engine by forcing that to be installed. (It requires that they are compatible.)

To sum it up, it is already possible to achieve what @mattias-p, but the current release process adds a push to update Engine and Zonemaster-LDNS when CLI or Backend is updated.
